### ISSUE TITLE
Output of accessor function as `NumericList` ?

### ISSUE THREAD

**Issue by philouail** [rformassspectrometry/Chromatograms#20]
If we follow the *Spectra* package structure the accessor function `rtime()` and `intensity()` should give an ouput as a `NumericList`

I always will check in the `validPeaksData()` function that we have numeric data anyway so is it necessary ? `NumericList` are a bit less efficient. 

just wondering if there is any reason to keep it that way and what would be the drawbacks to just have a `list` ? 
Could a future `ChromBackend` implementation need this ?

**Comment by jorainer** [comment]
I have no strong opinion here - with a slight preference to use `list` of `numeric` instead of `NumericList`.

**Comment by philouail** [comment]
![image](https://github.com/user-attachments/assets/d0c24eca-6d0e-461e-8020-910c8dfbe1c6)

Some extra info: here are the results of a microbenchmark with and without having a NumericList as an output instead of normal list output.... It's not just a bit less efficient, this is pretty consequential.

**Comment by jorainer** [comment]
I would go for a `list` of `numeric` for now. The backend needs to ensure that the `list` it returns is indeed a `list` of `numeric` (not some random `logical` or `character` in it), but that's something that the backend can check e.g. when initialized.

Worst case we can always wrap an additional `NumericList()` call around the returned `list`... 

it is just important that we properly define and document the (expected) type of the return value of methods (this should go into the documentation of the `ChromBackend`.

**Comment by philouail** [comment]
Yes as of now I find it more easy to implement and it seems that it is just more efficient in general. Also I implement a .CORE_PEAKS_VARIABLES which defines the names and class of these, which makes it easy for me to check that they are indeed numeric. 

If something comes up I'll add to the discussion :)

**Comment by jorainer** [comment]
Excellent, yes, I agree to split/distinguish between *core chromatogram variables* and *core peaks variables*.

**Comment by philouail** [comment]
We went with `list` as this system is not necessary anymore as we now completely separate peaksData and chromData. + the efficient is consequential as Johannes showed.
